src/components/tests/SourcesTree.spec.js

```diff
@@ -6,32 +6,617 @@ import React from "react";
 import { shallow } from "enzyme";
 import SourcesTree from "../../components/PrimaryPanes/SourcesTree";
 import * as I from "immutable";
+import { showMenu } from "devtools-contextmenu";
+import { copyToTheClipboard } from "../../utils/clipboard";
+import {
+  getShownSource,
+  getSelectedSource,
+  getDebuggeeUrl,
+  getExpandedState,
+  getProjectDirectoryRoot,
+  getSources
+} from "../../selectors";
+
+jest.mock("devtools-contextmenu", () => ({ showMenu: jest.fn() }));
+jest.mock("../../utils/clipboard", () => ({ copyToTheClipboard: jest.fn() }));
+jest.mock("../../selectors");
+
+const defaultSources = I.Map({
+  "server1.conn13.child1/39": createMockSource(
+    "server1.conn13.child1/39",
+    "http://mdn.com/one.js"
+  ),
+  "server1.conn13.child1/40": createMockSource(
+    "server1.conn13.child1/40",
+    "http://mdn.com/two.js"
+  ),
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockSource = I.Map({
+  "server1.conn13.child1/41": createMockSource(
+    "server1.conn13.child1/41",
+    "http://mdn.com/three.js"
+  )
+});
+
+const singleMockItem = createMockItem(
+  "http://mdn.com/one.js",
+  "one.js",
+  I.Map({ id: "server1.conn13.child1/39" })
+);
+
+const blackboxedMockItem = createMockItem(
+  "http://mdn.com/blackboxed.js",
+  "blackboxed.js",
+  I.Map({ id: "server1.conn13.child1/59" })
+);
+
+const blackboxedMockSource = I.Map({
+  "server1.conn13.child1/59": createMockSource(
+    "server1.conn13.child1/59",
+    "http://mdn.com/blackboxed.js",
+    true
+  )
+});
+
+const mockRootDirectory = {
+  contents: [],
+  name: "root",
+  path: "root/"
+};
+
+const mockDirectory = {
+  contents: [],
+  name: "folder",
+  path: "folder/"
+};
+
+const mockEvent = {
+  preventDefault: jest.fn(),
+  stopPropagation: jest.fn()
+};
+
+let defaultComponent;
+let defaultProps;
+let defaultState;
+let componentWithRoot;
+let propsWithRoot;
+let emptyComponent;
+let componentWithMdnRoot;
+let propsWithMdnRoot;
+let blackboxedComponent;
+
+describe("SourcesTree", () => {
+  beforeEach(() => {
+    let rendered = render();
+    defaultComponent = rendered.component;
+    defaultProps = rendered.props;
+    defaultState = defaultComponent.state();
+
+    rendered = render({
+      projectRoot: "root/"
+    });
+    componentWithRoot = rendered.component;
+    propsWithRoot = rendered.props;
+
+    emptyComponent = render({
+      projectRoot: "custom",
+      sources: I.Map()
+    }).component;
+
+    rendered = render({
+      projectRoot: "mdn"
+    });
+    componentWithMdnRoot = rendered.component;
+    propsWithMdnRoot = rendered.props;
+
+    blackboxedComponent = render({
+      sources: blackboxedMockSource
+    }).component;
```

Comment:
a lot of the setup being done here is not necessary for every test. the only shared element is `defaultComponent`. Also, the previous naming was a bit better. I think this is due to the misunderstanding of what I meant when i said colocate -- generally it is better to keep a reader from going up and down the code. It is hard to keep your mental model clear. Unless this setup is really needed for all the tests, then we are doing unnecessary work here computationally, and we are doing unnecessary work here mentally. 

so, once again -- try to keep variables in the test that you use them in. do not define extra stuff that is used in only one test in the before each. Given that we have the render function which will define everything that we need in a single like, it is unlikely that you need this before each at all. If this is confusing, please let me know, I will try to clarify. (also on slack you can find me and ask -- look for yulia)